src/sentry/api/event_search.py

```diff
@@ -536,7 +541,7 @@ def visit_boolean_filter(self, node, children):
                     search_key,
                     sep,
                     "=",
-                    search_value,
+                    [search_value.text, ""],
```

Comment:
boolean search_value needs to be made to look like a numeric value (with the unit) if we're going to be calling other visitors like this

Reply:
I kind of hate that we call visitors directly like this, just ends up being fragile to work with

Reply:
definitely agree. I think we should just extract the logic out for these visitors and have it do it here instead of trying to be clever. I'll try and clean that up in a follow up